src/app/Dashboard/AddCard.tsx

```diff
@@ -157,3 +227,166 @@ export const AddCard: React.FunctionComponent<AddCardProps> = (props: AddCardPro
     </>
   );
 };
+
+interface PropsConfigFormProps {
+  cardTitle: string;
+  controls: PropControl[];
+  config: any;
+  onChange: ({}) => void;
+}
+
+const PropsConfigForm = (props: PropsConfigFormProps) => {
+  const handleChange = React.useCallback(
+    (k) => (e) => {
+      const copy = { ...props.config };
+      copy[k] = e;
+      props.onChange(copy);
+    },
+    [props.config, props.onChange]
+  );
+
+  const handleNumeric = React.useCallback(
+    (k) => (e) => {
+      const value = (e.target as HTMLInputElement).value;
+      const copy = { ...props.config };
+      copy[k] = value;
+      props.onChange(copy);
+    },
+    [props.config, props.onChange]
+  );
+
+  const handleNumericStep = React.useCallback(
+    (k, v) => (e) => {
+      const copy = { ...props.config };
+      copy[k] = props.config[k] + v;
+      props.onChange(copy);
+    },
+    [props.config, props.onChange]
+  );
+
+  const createControl = React.useCallback(
+    (ctrl: PropControl): JSX.Element => {
+      let input: JSX.Element;
+      switch (ctrl.kind) {
+        case 'boolean':
+          input = (
+            <Switch label="On" labelOff="Off" isChecked={props.config[ctrl.key]} onChange={handleChange(ctrl.key)} />
+          );
+          break;
+        case 'number':
+          input = (
+            <NumberInput
+              inputName={ctrl.name}
+              inputAriaLabel={`${ctrl.name} input`}
+              value={props.config[ctrl.key]}
+              onChange={handleNumeric(ctrl.key)}
+              onPlus={handleNumericStep(ctrl.key, 1)}
+              onMinus={handleNumericStep(ctrl.key, -1)}
+            />
+          );
+          break;
+        case 'string':
+          input = (
+            <TextInput
+              type="text"
+              aria-label={`${ctrl.key} input`}
+              value={props.config[ctrl.key]}
+              onChange={handleChange(ctrl.key)}
+            />
+          );
+          break;
+        case 'text':
+          input = (
+            <TextArea
+              type="text"
+              aria-label={`${ctrl.key} input`}
+              value={props.config[ctrl.key]}
+              onChange={handleChange(ctrl.key)}
+            />
+          );
+          break;
+        case 'select':
+          input = (
+            <SelectControl handleChange={handleChange(ctrl.key)} config={props.config[ctrl.key]} control={ctrl} />
```

Comment:
Should be named `selectedConfig`?

Reply:
Not sure which part you're referring to

Reply:
Ah sorry the prop `config`?

Reply:
Err... the prop's name is `config`

https://github.com/cryostatio/cryostat-web/pull/747/files#diff-05851ec53c3e2fc6b4eaf23d7057435b91f0fde619760e9c319bd79ba0fe05f2R340

Reply:
Ahhh I mean since that prop is used in `selections` prop of the `Select` menu, maybe we should called it `selectedConfig`?